scio-google-cloud-platform/src/main/scala/com/spotify/scio/bigquery/MockBigQuery.scala

```diff
@@ -64,6 +64,32 @@ class MockBigQuery private (private val bq: BigQuery) {
     new MockTable(bq, t.getSchema, original, temp)
   }
 
+  /**
+   * Mock a BigQuery wildcard table using the table prefix and the suffix matched by the wildcard.
+   * Each table can be mocked only once in a test class.
```

Comment:
It is hard to separate the prefix from the suffix in the API since we need to check if the original table exists.
I kept the same API with `def mockWildcardTable(prefix: String, suffix: String)` but it is now possible to call it several times.

Moreover, having `withData(suffix1, data1)` will force to create a table on the fly, and doesn't allign well with the original API. We should have had smth like `mockTableData(original, data)` directly on `MockBigQuery` to be aligned.

We could imagine smth like `mockWildcard(prefix).mockTable(suffix1).withData(data1)` but I feel this indirection isn't helping

Reply:
IMO simplicity is a benefit here, so keeping a single `mockWildcardTable` call, analogous to the existing `mockTable`, is better than adding complexity. Most users will likely have a map of suffix -> data anyway, so they'll just be `foreach`ing over it.